File: julia/lusolve.jl
include("trianginf.jl")
include("triangsup.jl")
include("ludecomp.jl")

function luSolve(A::Matrix{T}, b::Vector{T}) where T <: Number
    luA = luDecomp(A)
    L = luA[1]
    U = luA[2]
    y = triangInf(L,b)
    x = triangSup(U,y)
    return x
end

A = [1. 1. -1.; 2. -1. 1.; -1. 2. 2.]
b = [-2., 5., 1.]

x = luSolve(A,b)
print(x)

File: lusolve.py
from numpy import zeros,array
from trianginf import trianginf
from triangsup import triangsup
from ludecomp import ludecomp

def lusolve(A,b):
   [L, U]=ludecomp(A)
   y=trianginf(L,b)
   x=triangsup(U,y)
   return x

# x=lusolve(array([[1,1,-1],[2,-1,1],[-1,2,2]]), array([-2, 5, 1]))
# print(x)